mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala

```diff
@@ -91,8 +130,16 @@ final class ChiSqSelector @Since("1.6.0") (@Since("1.6.0") override val uid: Str
         case Row(label: Double, features: Vector) =>
           OldLabeledPoint(label, OldVectors.fromML(features))
       }
-    val chiSqSelector = new feature.ChiSqSelector($(numTopFeatures)).fit(input)
-    copyValues(new ChiSqSelectorModel(uid, chiSqSelector).setParent(this))
+    var model = $(selectorType) match {
+      case "KBest" =>
+        new feature.ChiSqSelector().setNumTopFeatures($(numTopFeatures)).fit(input)
+      case "Percentile" =>
+        new feature.ChiSqSelector().setPercentile($(percentile)).fit(input)
+      case "Fpr" =>
+        new feature.ChiSqSelector().setAlpha($(alpha)).fit(input)
+      case _ => throw new IllegalStateException("Unknown ChiSqSelector Type.")
```

Comment:
Put the invalid type itself into the error, and you can use string interpolation here.